src/algorithm/radix4.rs

```diff
@@ -147,30 +136,33 @@ boilerplate_fft_oop!(Radix4, |this: &Radix4<_>| this.len);
 // Preparing for radix 4 is similar to a transpose, where the column index is bit reversed.
 // Use a lookup table to avoid repeating the slow bit reverse operations.
 // Unrolling the outer loop by a factor 4 helps speed things up.
-pub fn bitreversed_transpose<T: Copy>(
-    height: usize,
-    input: &[T],
-    output: &mut [T],
-    shuffle_map: &[usize],
-) {
-    let width = shuffle_map.len();
+pub fn bitreversed_transpose<T: Copy>(height: usize, input: &[T], output: &mut [T]) {
+    let width = input.len() / height;
+    let quarter_width = width / 4;
+
+    let rev_digits = (width.trailing_zeros() / 2) as usize;
+
     // Let's make sure the arguments are ok
     assert!(input.len() == output.len());
-    assert!(input.len() == height * width);
-    for (x, x_rev) in shuffle_map.chunks_exact(4).enumerate() {
+    for x in 0..quarter_width {
         let x0 = 4 * x;
         let x1 = 4 * x + 1;
         let x2 = 4 * x + 2;
         let x3 = 4 * x + 3;
 
+        let x_rev = [
```

Comment:
It could make sense to make a "reverse_bits_of_four" function that reverses four numbers in the same loop. I'm guessing that would make it good for the auto-vectorizer.

Reply:
Nope, not a good idea. That actually runs a tiny bit slower for some odd reason.

Reply:
Update on that. I can't measure any difference so it's not slower. And also not faster. Let's not bother.